js/ui/calendar/ui.calendar.js

```diff
@@ -243,28 +258,35 @@ const Calendar = Editor.inherit({
         return dateSerialization.getDateSerializationFormat(value);
     },
 
-    _convertToDate: function(value, optionName) {
+    _convertToDate: function(value) {
         return dateSerialization.deserializeDate(value);
     },
 
     _dateValue: function(value, event) {
+        const optionName = Array.isArray(value) ? 'values' : 'value';
         if(event) {
             if(event.type === 'keydown') {
                 const cellElement = this._view._getContouredCell().get(0);
                 event.target = cellElement;
             }
             this._saveValueChangeEvent(event);
         }
-        this._dateOption('value', value);
+        this._dateOption(optionName, value);
     },
 
     _dateOption: function(optionName, optionValue) {
         if(arguments.length === 1) {
-            return this._convertToDate(this.option(optionName), optionName);
+            return optionName === 'values' ?
+                (this.option('values') || []).map((value) => this._convertToDate(value)) :
```

Comment:
Let's create a separate variable to simplify this expression and the expression below.
```javascript
                const values = this.option('values') ?? [];
```

Reply:
done